lib/BackgroundJob/QuotaJob.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2023 Anna Larch <anna.larch@gmx.net>
+ *
+ * @author Anna Larch <anna.larch@gmx.net>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Mail\BackgroundJob;
+
+use OCA\Mail\Account;
+use OCA\Mail\Contracts\IMailManager;
+use OCA\Mail\Service\AccountService;
+use OCP\AppFramework\Db\DoesNotExistException;
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\BackgroundJob\IJobList;
+use OCP\BackgroundJob\TimedJob;
+use OCP\IUserManager;
+use OCP\Notification\IManager;
+use Psr\Log\LoggerInterface;
+use function sprintf;
+
+class QuotaJob extends TimedJob {
+	private IUserManager $userManager;
+	private AccountService $accountService;
+	private IMailManager $mailManager;
+	private LoggerInterface $logger;
+	private IJobList $jobList;
+	private IManager $notificationManager;
+
+	public function __construct(ITimeFactory $time,
+		IUserManager $userManager,
+		AccountService $accountService,
+		IMailManager $mailManager,
+		IManager $notificationManager,
+		LoggerInterface $logger,
+		IJobList $jobList) {
+		parent::__construct($time);
+
+		$this->userManager = $userManager;
+		$this->accountService = $accountService;
+		$this->logger = $logger;
+		$this->jobList = $jobList;
+		$this->mailManager = $mailManager;
+
+		$this->setInterval(24 * 60 * 60);
```

Comment:
IMO doing this less often would be fine too. The quota won't grow so rapidly. If it does, we won't catch it in time anyway.